### DISCUSSION TITLE
How can a pod know if itself is the new pod in the middle of the argo rollout?

### DISCUSSION THREAD

**Discussion by yidongw** [argoproj/argo-rollouts#2835] [Q&A]
Our pods are communicating with each other through Kafka messages. The requests start on the Web frontend. If the requests are coming from the preview frontend, the request will be labeled with the preview in the requests header. The receiving pod in the k8s will produce a kafka message with either preview label or no preview label.

Now the consumer pods need to decide if they want to consume this kafka message or not.

If consumer replica set has no preview pod (there is no update in the consumer Argo Rollout), the current active consumer pod will consume the kafka message regardless of having preview label or not.
If consumer replica set has preview pod (there is some update in the consumer Argo Rollout), the new consumer pod will only consume the kafka message with preview label on it; the old active consumer pod will only consume the kafka message with NO preview label on it.
So how can a pod know if itself is the new pod in the middle of the argo rollout? How do we usually handle this case?

**Comment by zachaller** [comment]
Maybe answered here: https://cloud-native.slack.com/archives/C01U781DW2E/p1686576619237719

**Reply by kamilglod** [reply]
I'm also interested in this thread, could you provide some answer that it's publicly available?

**Reply by zachaller** [reply]
<img width="624" alt="Screenshot 2023-08-23 at 9 57 35 AM" src="https://github.com/argoproj/argo-rollouts/assets/201390/7606a3bd-36d2-43a4-bcc5-2c7870bc8d0e">

**Reply by luke-quest** [reply]
Thanks for all this info.

For anyone else that comes across this page to save the 5 mins working out that the labels don't exist and you will also need to add the metadata label to your strategy like this (for bluegreen):
```
  strategy:
    blueGreen: 
      activeMetadata:
        labels:
          "release-stage": active
      previewMetadata:
        labels:
          "release-stage": preview
```

https://argo-rollouts.readthedocs.io/en/stable/features/ephemeral-metadata/

**Reply by kostis-codefresh** [reply]
I have written a longer article here about using ephemeral labels plus the downward api https://codefresh.io/blog/progressive-delivery-for-stateful-services-using-argo-rollouts/
